scripts/lua2dox.lua

```diff
@@ -58,7 +58,11 @@ local TYPES = { 'integer', 'number', 'string', 'table', 'list', 'boolean', 'func
 local TAGGED_TYPES = { 'TSNode', 'LanguageTree' }
 
 -- Document these as 'table'
-local ALIAS_TYPES = { 'Range', 'Range4', 'Range6', 'TSMetadata' }
+local ALIAS_TYPES = {
+  'Range', 'Range4', 'Range6', 'TSMetadata',
+  'vim.filetype.add.filetypes',
+  'vim.filetype.match.args'
```

Comment:
What's the plan here? This doesn't look sustainable. Why not just use `table` ?

Reply:
We want to use actual non-table types in the code so we get good type checking, but for now we render these as table in the docs.

In the future it would be good to automate doc generation of the types.